### ISSUE TITLE
Consolidated Fusion Screen

### ISSUE THREAD

**Issue by claytondaley** [swarfarm/swarfarm#121]
The current UI isn't great for fusing a nat5 because you have to jump around between the different 4* fusions.  I'm also (currently) fusing a bunch of random stuff for different reasons (Lushen/KFG skillups, Harp Magician to use, and pretty much any other nat4 to convert to LS pieces).

For a single nat5 it'd be really nice to have a "consolidated" fusion view e.g. a hierarchical table:

Veromos
- Undine                        [source] [units]  
  - Water Bearman         [source] [units]  [pieces]
  - Water Grim Reaper   [source] [units]  [pieces]
  - ...

Right now, I'd also benefit from a rollup of all of the fusion units kinda like this guy but vertical:

![image](https://user-images.githubusercontent.com/1097439/72642079-2a0e9180-3931-11ea-8241-abfbd09d2698.png)

**Comment by deirdresm** [comment]
Concur. I've actually been thinking about this.

I'd also like to add that I'd like to see which you're closest to fusing.

So I'd add a number box in between the fused monster icon and the components that's the % done, probably 125% - 150% size.

Now…as to calculating how "done" something is….

1. Experience. Obviously a monster you don't have counts as zero, and then you've got to count all the grades (stars) and experience for each grade vs. how much they need to have in total.
2. Essence needed. I think counting essence in fusion terms (e.g., 100 low = 10 medium = 1 high) is fair, and since you at most have to wait a week to fuse (one hopes), count each element the same as magic.
3. Mana needed. This seems pointless, but just thought I'd raise it as something I'd discarded. Most people's mana changes often enough to make tracking it meaningless, I think.

So the real question is: when calculating how "done" something is, how do you count the relative value of levels vs. essence? Some days, the essence is harder.  (I swear, I spent three full refills on Magic B10 the other day and got zero high essence but 72 low. Grr.)

I'll get some data together and put it in another comment in a bit about experience per level as it's asymmetric.

Edit: also, rather than make it super-wide, it might make more sense if it were broken per three 5* fusions. That would allow more room for some info about the 4* stuff.

**Comment by claytondaley** [comment]
Completion %s is an interesting upgrade but shouldn't block the UI itself.  In theory swarfarm has the data to do the % completion calculation in energy.

 - The essence dungeons can tell you how many runs need to generate the required essences (or their fusion materials).  You'd want to calculate the magic essences first and estimate the net-elemental essences gained before calculating the elemental essence runs.
 - Scenarios can tell you how much energy it takes to get the actual fusion units to the appropriate level (I assume you use a benchmark like faimon).
 - The system can even estimate the cost to farm farmable or acquire SD units
 - In theory, it can also provide information about food generated by both processes (or required in addition).  But how do you discriminate food available for fusion from food mentally allocated to e.g. a 6*.

I don't object to the idea (and energy is a natural, unifying currency) but I wouldn't want the complexity of this stage to dissuade the consolidated UI.

**Comment by deirdresm** [comment]
Fair enough. I have ADHD, and my personal design goal would be a reminder to me of what I'm working on. Or, that I've wound up with something that's easy to assemble. I find SW a game where it's particularly difficult to keep focus on what it is I want to accomplish once I got past the onboarding.

One other UI thing I think might be an enhancement: where the 4* units have enough essence to awaken, put a glow around them so they're more obvious. Obviously that'll be an issue if you have enough for three of the five possible six 5* fusions, but I'm not sure there's a good solution for that.

On Faimon/SD energy, there is some RNG involved, both in the levels of the monsters, as well as the distribution of them. In Faimon, sometimes you get more Inugami, sometimes more Hellhounds. Also, if you're using a rep, that'll affect experience (because you do not count the rep when dividing the experience like you would your own farming unit). Also, if you're trying to get one monster just that little last way to the next level, it's best to have it be the ONLY monster and use a rep so it doesn't share the experience with anyone else.

**Comment by PeteAndersen** [comment]
I've also been thinking about an improved fusion screen. The rollup/summary page is definitely useful. I'd like to have a non-personalized fusion recipe screen for public reference. 

I am all for everything discussed so far but I think adding details like # of runs to level or gain essences might be something that can be implemented later. The effort is high and integrating this much detail might be difficult to present well. The leveling up part can be tied in with #108 if/when that ever happens.

On completion: The time and effort involved in acquiring a monster, leveling it, and gathering essences for awakening also isn't equal so I feel like a % completion is hard to define and is more likely incorrect or misleading than useful. It might be better defined as steps or checkpoints to complete. Not as a progress bar, though, because they can be completed in any order.

**Comment by deirdresm** [comment]
True. I actually had been planning to do some of this for an iOS app for myself so I could tinker without bothering anyone.

I do like the existing setup that checks whether you’ve leveled, awakened, etc., so I wouldn’t want to lose what works about that.
